packages/palette/src/elements/Tabs/Tabs.tsx

```diff
@@ -38,39 +45,48 @@ export const useTabs = ({
   initialTabIndex = 0,
   onChange,
 }: TabsProps) => {
-  const tabs = flattenChildren<TabLike>(children).map((child) => ({
-    child,
-    ref: createRef<HTMLButtonElement | null>(),
-  }))
+  const tabs = useMemo(
+    () =>
+      flattenChildren<TabLike>(children).map((child) => ({
+        child,
+        ref: createRef<HTMLButtonElement | null>(),
+      })),
+    [children]
+  )
 
   const [activeTabIndex, setActiveTabIndex] = useState<number>(initialTabIndex)
-  const activeTab = tabs[activeTabIndex]
+
+  // Wraps active tab in a ref to prevent re-rendering.
+  // This means that we need to update the active tab before the index state change
+  // to catch the re-render.
+  const activeTab = useRef(tabs[activeTabIndex])
```

Comment:
This is a little annoying but is how we can prevent scrolls from happening on arbitrary re-renders. We just need to update this ref's value before we update the `activeTabIndex` so that updates catch the render.